cmd/sovereignnode/sovereignNodeRunner.go

```diff
@@ -0,0 +1,1715 @@
+	"github.com/multiversx/mx-chain-go/facade"
+	"github.com/multiversx/mx-chain-go/facade/initial"
+	mainFactory "github.com/multiversx/mx-chain-go/factory"
+	apiComp "github.com/multiversx/mx-chain-go/factory/api"
+	bootstrapComp "github.com/multiversx/mx-chain-go/factory/bootstrap"
+	consensusComp "github.com/multiversx/mx-chain-go/factory/consensus"
+	coreComp "github.com/multiversx/mx-chain-go/factory/core"
+	cryptoComp "github.com/multiversx/mx-chain-go/factory/crypto"
+	dataComp "github.com/multiversx/mx-chain-go/factory/data"
+	heartbeatComp "github.com/multiversx/mx-chain-go/factory/heartbeat"
+	networkComp "github.com/multiversx/mx-chain-go/factory/network"
+	processComp "github.com/multiversx/mx-chain-go/factory/processing"
+	stateComp "github.com/multiversx/mx-chain-go/factory/state"
+	statusComp "github.com/multiversx/mx-chain-go/factory/status"
+	"github.com/multiversx/mx-chain-go/factory/statusCore"
+	"github.com/multiversx/mx-chain-go/genesis"
+	"github.com/multiversx/mx-chain-go/genesis/parsing"
+	"github.com/multiversx/mx-chain-go/health"
+	"github.com/multiversx/mx-chain-go/node"
+	"github.com/multiversx/mx-chain-go/node/metrics"
+	"github.com/multiversx/mx-chain-go/outport"
+	"github.com/multiversx/mx-chain-go/p2p"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/process/interceptors"
+	"github.com/multiversx/mx-chain-go/sharding/nodesCoordinator"
+	"github.com/multiversx/mx-chain-go/state/syncer"
+	"github.com/multiversx/mx-chain-go/storage/cache"
+	storageFactory "github.com/multiversx/mx-chain-go/storage/factory"
+	"github.com/multiversx/mx-chain-go/storage/storageunit"
+	trieFactory "github.com/multiversx/mx-chain-go/trie/factory"
+	trieStatistics "github.com/multiversx/mx-chain-go/trie/statistics"
+	"github.com/multiversx/mx-chain-go/trie/storageMarker"
+	"github.com/multiversx/mx-chain-go/update/trigger"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("sovereignNode")
+
+const (
+	// TODO: remove this after better handling VM versions switching
+	// defaultDelayBeforeScQueriesStartInSec represents the default delay before the sc query processor should start to allow external queries
+	defaultDelayBeforeScQueriesStartInSec = 120
+
+	maxTimeToClose = 10 * time.Second
+	// SoftRestartMessage is the custom message used when the node does a soft restart operation
+	SoftRestartMessage = "Shuffled out - soft restart"
+)
+
+// sovereignNodeRunner holds the sovereign node runner configuration and controls running of a node
+type sovereignNodeRunner struct {
+	configs *config.Configs
+}
+
+// NewSovereignNodeRunner creates a sovereignNodeRunner instance
+func NewSovereignNodeRunner(cfgs *config.Configs) (*sovereignNodeRunner, error) {
+	if cfgs == nil {
+		return nil, fmt.Errorf("nil configs provided")
+	}
+	return &sovereignNodeRunner{
+		configs: cfgs,
+	}, nil
+}
+
+func (snr *sovereignNodeRunner) getConsensusModel() consensus.ConsensusModel {
```

Comment:
also here, I do not think we gonna need these 2 functions.

Reply:
Right, good suggestion. Applied it